server/appender_test.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"testing"
+	"time"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/mattermost/mattermost-plugin-metrics/server/mocks"
+)
+
+func nopMutator(l labels.Labels) labels.Labels { return l }
+
+func TestAppendFn(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	ma := mocks.NewMockAppender(ctrl)
+	ml := mocks.NewMockLogger(ctrl)
+	lg := &metricsLogger{api: ml}
+
+	cf := &configuration{}
+	cf.SetDefaults()
+
+	sampleMutator = nopMutator
+
+	t.Run("empty body", func(t *testing.T) {
+		ch := newScrapeCache()
+		err := appendFn(ma, lg, ch, []byte{}, "application/openmetrics-text", time.Now(), cf)
+		require.EqualError(t, err, "data does not end with # EOF")
+	})
+
+	t.Run("metric group", func(t *testing.T) {
+		ch := newScrapeCache()
+		body := `# TYPE foo_seconds counter
+# UNIT foo_seconds seconds
+foo_seconds_count 0
+foo_seconds_sum 1
+# EOF
+`
+		ts := time.Now()
+		ls := labels.Labels{{Name: "__name__", Value: "foo_seconds_count"}}
+		ma.EXPECT().Append(storage.SeriesRef(0), ls, ts.UnixMilli(), float64(0)).Times(1)
+		ls = labels.Labels{{Name: "__name__", Value: "foo_seconds_sum"}}
+		ma.EXPECT().Append(storage.SeriesRef(0), ls, ts.UnixMilli(), float64(1)).Times(1)
+		err := appendFn(ma, lg, ch, []byte(body), "application/openmetrics-text", ts, cf)
+		require.NoError(t, err)
+	})
+
+	t.Run("metric histogram", func(t *testing.T) {
+		ch := newScrapeCache()
+		body := `# TYPE foo histogram
+foo_bucket{le="0.0"} 0
+foo_bucket{le="0.1"} 1
+# EOF
+`
+		ts := time.Now()
+		ls := labels.Labels{
+			{Name: "__name__", Value: "foo_bucket"},
+			{Name: "le", Value: "0.0"},
+		}
+		ma.EXPECT().Append(storage.SeriesRef(0), ls, ts.UnixMilli(), float64(0)).Times(1)
```

Comment:
I don't understand this. As per the code, it's supposed to call `AppendHistogram` and not `Append` right? If this is right, can we add a test to make it call `AppendHistogram` as well?

Reply:
AppendHistogram is being used only for the Protobuf data type, for openmetrics-text it's always the regular append.